launcher/ui/MainWindow.cpp

```diff
@@ -2269,10 +2269,25 @@ void MainWindow::on_actionCreateInstanceShortcut_triggered()
             QMessageBox::critical(this, tr("Create instance shortcut"), tr("Failed to create icon for shortcut."));
             return;
         }
-        
-        if (FS::createShortcut(FS::PathCombine(desktopPath, m_selectedInstance->name()),
-                           appPath, { "--launch", m_selectedInstance->id() },
-                           m_selectedInstance->name(), iconPath)) {
+
+        auto desktopFilePath = FS::PathCombine(desktopPath, m_selectedInstance->name() + ".desktop");
+        QStringList args;
+        if (DesktopServices::isFlatpak()) {
+            auto fileDialog = new QFileDialog();
```

Comment:
does this need to be a pointer? if not, make it a variable on the stack to not leak memory